sublime_jedi/completion.py

```diff
@@ -146,61 +154,69 @@ def on_query_completions(self, prefix, locations):
         logger.info('JEDI completion triggered.')
 
         settings = get_settings(self.view)
-        only_jedi_completion = (
-            settings['sublime_completions_visibility'] in ('default', 'jedi')
-        )
-
-        previous_char = self.view.substr(locations[0] - 1)
         if settings['only_complete_after_regex']:
+            previous_char = self.view.substr(locations[0] - 1)
             if not re.match(settings['only_complete_after_regex'], previous_char):
                 return False
 
-        cplns = ask_daemon_with_timeout(
-            self.view,
-            'autocomplete',
-            location=locations[0],
-            timeout=settings['completion_timeout']
-        )
-
-        logger.info("Completion completed.")
-
-        cplns = [tuple(x) for x in self._sort_completions(cplns)]
-        logger.debug("Completions: {0}".format(cplns))
-
-        # disabled due to can't reproduce
-        # self._fix_tab_completion_issue()
-
-        if only_jedi_completion:
-            return cplns, PLUGIN_ONLY_COMPLETION
-        return cplns
+        with self._lock:
```

Comment:
throttling?

Reply:
Some times two threads enter here at the same time accessing the completion data structures, this is to prevent this access from blowing up.